packages/passage/src/stimulus-tabs.jsx

```diff
@@ -75,6 +75,37 @@ class StimulusTabs extends React.Component {
     });
   };
 
+  handleKeyDown = (event, currentTabId) => {
+    const { key } = event;
+    const { tabs } = this.props;
+  
+    let newTabIndex = -1;
+  
+    if (['ArrowRight', 'ArrowDown', 'ArrowLeft', 'ArrowUp'].includes(key)) {
+      event.preventDefault();
+      const currentIndex = tabs.findIndex((tab) => tab.id === currentTabId);
+  
+      if (key === 'ArrowRight' || key === 'ArrowDown') {
+        newTabIndex = (currentIndex + 1) % tabs.length;
+      } else if (key === 'ArrowLeft' || key === 'ArrowUp') {
+        newTabIndex = (currentIndex - 1 + tabs.length) % tabs.length;
+      }
+    } else if (key === 'Home') {
+      event.preventDefault();
+      newTabIndex = 0;
+    } else if (key === 'End') {
+      event.preventDefault();
+      newTabIndex = tabs.length - 1;
+    } else if (key === 'Enter' || key === ' ') {
+      event.preventDefault();
+      newTabIndex = tabs.findIndex((tab) => tab.id === currentTabId);
+    }
+  
+    if (newTabIndex !== -1) {
+      this.handleChange(event, tabs[newTabIndex].id);
+    }
+  };
+  
```

Comment:
Couldn't we make this nicer with a switch? Also, it looks like we call event.preventDefault every time one if is called. Can we only call it if newTabIndex !== -1?

Reply:
sure, I added the switch and refactored the function.